paig-server/backend/paig/api/shield/scanners/BedrockGuardrailScanner.py

```diff
@@ -0,0 +1,105 @@
+import logging
+import boto3
+import os
+
+from api.shield.model.scanner_result import ScannerResult
+from api.shield.scanners.BaseScanner import Scanner
+
+logger = logging.getLogger(__name__)
+
+
+class BedrockGuardrailScanner(Scanner):
+    """
+    Scanner implementation for applying guard rail policies in the input prompt.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initialize the GuardrailScanner with the specified parameters.
+
+        Parameters:
+            **kwargs: keyword arguments passed from properties file.
+            E.g.
+            name (str): The name of the scanner.
+            request_types (list): List of request types that the scanner will handle.
+            enforce_access_control (bool): Flag to enforce access control.
+            model_path (str): Path to the model used by the scanner.
+            model_score_threshold (float): Threshold score for the model to consider a match.
+            entity_type (str): Type of entity the scanner is looking for.
+            enable (bool): Flag to enable or disable the scanner.
+        """
+        super().__init__(**kwargs)
+
+        self.guardrail_id, self.guardrail_version, self.region = self.get_guardrail_details()
+
+        self.bedrock_client = boto3.client(
+            'bedrock-runtime',
+            region_name=self.region
```

Comment:
We are implicitly supporting only IAM instance role - we may want to support access-key/secret-key/session-token as well?

Reply:
Will create a task for this

Reply:
we can resolve this after task is filed

Reply:
filed task https://github.com/privacera/paig/issues/122